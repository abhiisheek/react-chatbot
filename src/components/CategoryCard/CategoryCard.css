@import '../_variables.css';

.card {
  border: 1px solid black;
  display: flex;
  min-width: 80px;
  flex-direction: column;
  align-items: center;
  cursor: pointer;
  box-shadow: var(--shadow-3p);
  box-sizing: border-box;
}

.title {
  margin-bottom: 10px;
}

.icon {
  border-radius: 50%;
  border: 1px solid black;
  width: 40px;
  height: 40px;
  position: relative;
  top: -20px;
  z-index: 100;
  background: #ffffff;
  display: flex;
  justify-content: center;
  align-items: center;
}
