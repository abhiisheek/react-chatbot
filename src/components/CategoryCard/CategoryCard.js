import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import classNames from 'classnames';

import styles from './CategoryCard.css';

/** Component to show various categories the chatbot can handle in the form of clickable cards */
class CategoryCard extends PureComponent {
  handleOnClick = () =>
    this.props.onClick({
      name: this.props.name,
      key: this.props.value,
      shortName: this.props.shortName || this.props.name
    });

  render() {
    const {
      name,
      iconClass,
      bgColor,
      textColor,
      iconColor,
      className
    } = this.props;
    return (
      <div
        onClick={this.handleOnClick}
        className={classNames(className, styles.card)}
        style={{
          backgroundColor: bgColor,
          borderColor: bgColor,
          color: textColor
        }}
      >
        <div
          className={classNames(styles.icon, iconClass)}
          style={{ borderColor: bgColor, color: iconColor }}
        />
        <div className={styles.title}>{name}</div>
      </div>
    );
  }
}

CategoryCard.displayName = 'CategoryCard';

CategoryCard.propTypes = {
  /** Name of the Category */
  name: PropTypes.string.isRequired,
  /** Short name of the Category, which will be used to disply in header region on category selection */
  shortName: PropTypes.string,
  /** Unique id/type for the category */
  value: PropTypes.string.isRequired,
  /** Handler to handling category selection */
  onClick: PropTypes.func.isRequired,
  /** CSS Classname for category icon*/
  iconClass: PropTypes.string.isRequired,
  /** Color code for category icon */
  iconColor: PropTypes.string.isRequired,
  /** Custom CSS class for handling the card look */
  className: PropTypes.string,
  /** Background color code for the category card */
  bgColor: PropTypes.string,
  /** Color code for text */
  textColor: PropTypes.string
};

CategoryCard.defaultProps = {
  bgColor: '#ffffff',
  textColor: '#000000'
};

export default CategoryCard;
