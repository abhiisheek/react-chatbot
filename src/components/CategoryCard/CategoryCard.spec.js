import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import CategoryCard from './CategoryCard';

let handleOnCardClick;

describe('CategoryCard', () => {
  beforeEach(() => {
    handleOnCardClick = jest.fn();
  });

  it('Should have displayName set on the component', () => {
    expect(CategoryCard.displayName).toBeDefined();
  });

  it('Snapshot test', () => {
    const snapshot = renderer
      .create(
        <CategoryCard
          name="WISMO"
          value="gist_wismo"
          onClick={handleOnCardClick}
          iconClass="wismo_icon"
          iconColor="orange"
          bgColor="white"
          textColor="black"
        />
      )
      .toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('On card click, should call handleOnCardClick with name as shortName when shortName is not defined', () => {
    const appName = { name: 'WISMO', key: 'gist_wismo' };
    const wrapper = shallow(
      <CategoryCard
        name={appName.name}
        value={appName.key}
        onClick={handleOnCardClick}
        iconClass="wismo_icon"
        iconColor="orange"
        bgColor="white"
        textColor="black"
      />
    );
    wrapper.simulate('click');
    expect(handleOnCardClick).toHaveBeenCalled();
    expect(handleOnCardClick).toHaveBeenCalledWith({
      ...appName,
      shortName: appName.name
    });
  });

  it('On card click, should call handleOnCardClick', () => {
    const appName = { name: 'WISMO', key: 'gist_wismo', shortName: 'WISMO' };
    const wrapper = shallow(
      <CategoryCard
        name={appName.name}
        value={appName.key}
        onClick={handleOnCardClick}
        iconClass="wismo_icon"
        iconColor="orange"
        bgColor="white"
        textColor="black"
      />
    );
    wrapper.simulate('click');
    expect(handleOnCardClick).toHaveBeenCalled();
    expect(handleOnCardClick).toHaveBeenCalledWith({
      ...appName
    });
  });
});
