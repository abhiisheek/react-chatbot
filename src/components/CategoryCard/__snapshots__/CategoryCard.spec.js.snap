// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryCard Snapshot test 1`] = `
<div
  className="card"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "white",
      "borderColor": "white",
      "color": "black",
    }
  }
>
  <div
    className="icon wismo_icon"
    style={
      Object {
        "borderColor": "white",
        "color": "orange",
      }
    }
  />
  <div
    className="title"
  >
    WISMO
  </div>
</div>
`;
